src/App/App.js

```diff
@@ -52,13 +53,15 @@ class App extends Component {
     if (this.state.user.name) {
       return <button className='log-button' onClick={this.logOutUser}>Logout</button>
     } else if (this.state.hasLoginView) {
-      return 
+      return
     }
     else {
       return <Link to="/login"><button className='log-button'>Login</button></Link>
     }
   }
 
+
+
   render() {
     return (
       <main>
```

Comment:
In <Header /> - you're passing down the function determineHeaderText. Do you perhaps mean to be passing down the header text itself? Like ... `<Header headerText={this.determineHeaderText()} logButtonStatus={this.determineLogButtonStatus()} />` since both those functions depend on the App's state? That way, the Header won't have to do the logic of INVOKING those functions, it can just use the output those functions produce.

Reply:
Love this idea, Leta!